tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs

```diff
@@ -0,0 +1,227 @@
+            A.CallTo(() => _fakeAzureTableReaderWriter.UpdateUnitPriceChangeStatusEntity(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).MustHaveHappened();
+            A.CallTo(() => _fakeAzureTableReaderWriter.AddUnitPriceChangeEntity(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).MustHaveHappened();
+
+            A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
+            && call.GetArgument<LogLevel>(0) == LogLevel.Information
+            && call.GetArgument<EventId>(1) == EventIds.AppendingUnitofSalePricesToEncEventInWebJob.ToEventId()
+            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Appending UnitofSale prices to ENC event in webjob.").MustHaveHappened();
+
+            A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
+            && call.GetArgument<LogLevel>(0) == LogLevel.Information
+            && call.GetArgument<EventId>(1) == EventIds.DownloadBulkPriceInformationEventFromAzureBlob.ToEventId()
+            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Downloading Price Change information from blob").MustHaveHappened();
+
+            A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
+            && call.GetArgument<LogLevel>(0) == LogLevel.Information
+            && call.GetArgument<EventId>(1) == EventIds.UploadedSlicedEventInAzureBlob.ToEventId()
+            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Sliced event is uploaded in blob storage successfully.").MustHaveHappened();
+
+        }
+
+        [Test]
+        public void WhenUnitPriceEntitiesAreIncompleted_ThenShouldSliceAndPublish()
+        {
+            IList<PriceChangeMasterEntity> priceChangeMasterEntities = new List<PriceChangeMasterEntity>();
+            priceChangeMasterEntities.Add(new()
+            {
+                CorrId = "FakeCorrID",
+                ETag = new(),
+                PartitionKey = "FakeCorrID",
+                RowKey = "FakeCorrID",
+                Status = "Incomplete",
+                Timestamp = new DateTimeOffset()
+            });
+            IList<UnitPriceChangeEntity> unitPriceChangeEntities = new List<UnitPriceChangeEntity>();
+            unitPriceChangeEntities.Add(new UnitPriceChangeEntity()
+            {
+                Eventid = "FakeEventID",
+                MasterCorrid = "FakeCorrID",
+                RowKey = "FakeEventID",
+                ETag = new(),
+                PartitionKey = "FakeEventID",
+                Status = "Incomplete",
+                Timestamp = new DateTimeOffset(),
+                UnitName = "FakeUnitName"
+            });
+            unitPriceChangeEntities.Add(new UnitPriceChangeEntity()
```

Comment:
can be changed to - new UnitPriceChangeEntity()

Reply:
changed to new UnitPriceChangeEntity()